### PR TITLE
Fix #101: SARIF validator console output should be in standard VS format

### LINKED ISSUE #101: SARIF validator console output should be in standard VS format

### PR DESCRIPTION
**Pull request by lgolding**
The product functionality was checked in during the previous merge a3fb424, but we didn't close the issue then because we hadn't yet provided unit tests for the `FormatForVisualStudio` extension methods that we'd added to the Driver SDK to support the implementation. Now we add the tests and close the issue.

Also:
* Reimplement `GetFirstSentenceTests` as a `Theory`, which is much more compact.

@michaelcfanning

### CODE REVIEW

File: src/Sarif.FunctionalTests/Sarif.FunctionalTests.csproj [hunk 1]
```diff
@@ -343,6 +343,9 @@
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
+  </ItemGroup>
```
Comment (lgolding) [inline src/Sarif.FunctionalTests/Sarif.FunctionalTests.csproj]: Added automatically by VS. See this [VS Connect thread](https://connect.microsoft.com/VisualStudio/feedback/details/800245/vs2013rc-adds-to-vs2012-c-project-section-itemgroup) for explanation.

File: src/Sarif.Driver/Sdk/ExtensionMethods.cs [hunk 2]
```diff
@@ -84,7 +84,7 @@ public static string FormatForVisualStudio(this Result result, IRuleDescriptor r
                 var lastComponent = components.Last();
                 messageLines.Add(
                     string.Format(
-                        CultureInfo.InvariantCulture, "{0}:{1} {2} {3}: {4}",
+                        CultureInfo.InvariantCulture, "{0}{1}: {2} {3}: {4}",
```
Comment (lgolding) [inline src/Sarif.Driver/Sdk/ExtensionMethods.cs]: Fixed bug in implementation. Now I can run the validator from the VS Tools menu, double-click on an output line, and navigate to the site of the issue.

**Comment by lgolding** [comment]
@michaelcfanning merging this in; feel free to comment _post hoc_.

File: src/Sarif.Driver.UnitTests/Sdk/ExtensionMethodsTests.cs [hunk 3]
```diff
@@ -1,60 +1,26 @@
...
-        private readonly ITestOutputHelper _testOutputHelper;
-
-        public ExtensionMethodsTests(ITestOutputHelper output)
-        {
-            _testOutputHelper = output;
-        }
-
-        [Fact]
-        public void GetFirstSentenceTests()
+        [Theory]
```
Comment (michaelcfanning) [inline src/Sarif.Driver.UnitTests/Sdk/ExtensionMethodsTests.cs]: Thanks for this conversion. :)

File: src/Sarif.Driver.UnitTests/Sdk/FormatForVisualStudioTests.cs [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            null,           // options
+            new Dictionary<string, string>
+            {
+                [TestFormatSpecifier] = "First: {0}, Second: {1}"
+            },
+            null,           // helpUri
+            null,           // properties
+            null);          // tags
+
+        private static readonly Region MultiLineTestRegion = new Region
```
Comment (michaelcfanning) [inline src/Sarif.Driver.UnitTests/Sdk/FormatForVisualStudioTests.cs]: There are missing cases for formatting the region. Will these go in tests associated with the code that produces the (x,x) notation?

**Comment by michaelcfanning** [comment]
LGTM. We need more test cases that exercise different region possibilities, for example, same line, multicolumn

**Comment by lgolding** [comment]
We do have a test that exercises same line, multicolumn (see the declaration and usage of  `SingleLineTestRegion`). We need one for same line, single column though.

**Comment by michaelcfanning** [comment]
Here are the cases. The latter two are covered, yes?

            //    (startLine) Region() new { StartLine = 5 }
            //    (startLine-endLine) new Region {StartLine = 5, EndLine = 6 }
            //    (startLine,startColumn) new Region() { StartLine = 5, StartColumn = 2 }
            //    (startLine,startColumn-endColumn)
            //    (startLine,startColumn,endLine,endColumn)

**Comment by lgolding** [comment]
Yes, and I was just adding the middle one. I'll add the other two now as well. PR in a moment.

**Pull request merged**